validation-test/stdlib/Range.swift.gyb

```diff
@@ -403,6 +403,19 @@ ${TestSuite}.test("Equatable") {
   checkEquatable(instances, oracle: { $0 == $1 })
 }
 
+${TestSuite}.test("Hashable") {
+  let _1 = ${Bound}(1)
+  let _2 = ${Bound}(2)
+
+  let instances: [${Self}<${Bound}>] = [
+    _1${op}_1,
+    _1${op}_2,
+    _2${op}_2,
+  ]
+  // The following fails with error: ambiguous reference to member '=='
+  checkHashable(instances, equalityOracle: { $0 == $1 })
```

Comment:
There are two issues here, both of which cause `${Self}<${Bound}>` to not be `Hashable`:

- `${Self}` is `ClosedRange` in half of the cases, but it doesn't implement `Hashable` yet.  `ClosedRange` is now defined in `ClosedRange.swift` rather than gybbed out from the same template as `Range`, so we need to declare its conformance separately.
- `${Bound}` is either `MinimalComparableValue` or `MinimalStrideableValue`. Unfortunately, neither of these implement `Hashable`, so the conditional conformance won't apply. Adding `Hashable` conformance to these types wouldn't be a good idea, either. :-/ The range unification makes the `MinimalStrideableValue` case unnecessary here, so we can probably get away with defining a `MinimalComparableHashableValue` struct and run the test on that, probably in a separate gyb loop. (Yuck.)